src/components/tx-flow/flows/SignMessageOnChain/ReviewSignMessageOnChain.tsx

```diff
@@ -38,15 +39,19 @@ const ReviewSignMessageOnChain = ({ message, method, requestId }: SignMessageOnC
   const chainId = useChainId()
   const { safe } = useSafeInfo()
   const onboard = useOnboard()
+  const web3 = useWeb3()
   const { safeTx, setSafeTx, setSafeTxError } = useContext(SafeTxContext)
 
   useHighlightHiddenTab()
 
   const isTextMessage = method === Methods.signMessage && typeof message === 'string'
   const isTypedMessage = method === Methods.signTypedMessage && isObjectEIP712TypedData(message)
 
-  const readOnlySignMessageLibContract = useMemo(() => getReadOnlySignMessageLibContract(chainId), [chainId])
-  const signMessageAddress = readOnlySignMessageLibContract.getAddress()
+  const signMessageLibContract = useMemo(() => {
+    if (web3) {
+      return getSignMessageLibContract(chainId, web3)
```

Comment:
We should think about adding a getInterface / getDeployment function which only uses (abi + address) as this would be enough here and in many other components. Then we would not require `web3` at all. Only the `chainId` and Safe version.

Reply:
I saw this pattern emerge whilst removing the read only instances and totally agree. We could use the deployment getters from `safe-deployments` for this.

I would suggest this is done in a separate PR though so as to not increase the scope of this: the getters return contract instances or `undefined`. We'd need to `return` early or `throw` to ensure type safety. What do you think?

Reply:
I think that was the reason we added those readOnly instances of the contracts to not have a dependency to the provider. Are those contracts responsible for the read only RPC requests that @francovenica found?

Reply:
> I think that was the reason we added those readOnly instances of the contracts to not have a dependency to the provider.

This I'm not sure about, but we should certainly pivot to using the deployments repo imo.

> Are those contracts responsible for the read only RPC requests that @francovenica found?

They were indeed.